magento/ece-tools
# Added changes to allow db:status to determine if setup:upgrade is req…

### Description
During the Deploy Phase - Upgrade/Setup ALWAYS runs setup:upgrade.
It has been determined that the time overhead of running setup:db:status was average 5s but setup:upgrade was average 14s. This check will reduce the deploy phase by 11s

These tests were done on a Sandbox Pro-12 Environment - project ID z3ek236yy2yv2 and on the STAGING branch.

Whilst this will add up to 5s to a deploy that does require DB-Updates - these SHOULD be a) run separate to functional deploys b) be a much smaller percentage of deploys.

On the environment above the downtime for deploy was clocked at 13s (Maintenance On to Maintenance Off) this means sessions requests only being queued for that time. This is much more desirable than the previous 24s

### Fixed Issues (if relevant)
1. magento/ece-tools#419: [ECE-Tools] Conditionally Execute setup:upgrade During Deployment

### Manual testing scenarios
1. Generate a build that runs bin/magento setup:upgrade
2. SSH into server and run bin/magento setup:db:status
3. If DB reports that upgrade STILL needs running then ensure patch MDVA-21578_EE_2.3.2_COMPOSER_v1.patch is installed on project and repeat 1-3
4. Generate a build that requires a DB update - check logs - setup:db:status should be logged and then the setup:upgraded immediately after.
5. Generate a build that DOES NOT require a DB update - check logs - setup:db:status should be logged and then the setup:upgrade will be absent.
6. REPEAT STEP 4

### Contribution checklist
 - [X ] Pull request has a meaningful description of its purpose
 - [X ] All commits are accompanied by meaningful commit messages
 - [NA ] All new or changed code is covered with unit/integration tests (if applicable)
 - [NA ] All automated tests passed successfully (all builds on Travis CI are green)
